File: code/jvm/group7/src/main/kotlin/group7/daw/issue/models/Issue.kt
package group7.daw.issue.models

data class Issue(
        val user_id: Int,
        val project_id: Int,
        val issue_id: Int?,
        val name:String,
        val description: String,
        val opened_on: Int,
        val closed_on: Int?,
        val state_name: String
)

File: code/js/group7/src/common/types/issue.ts
import { ActionsSiren } from "./actions";
import { LinksSiren } from "./links";

export type IssueEntityProperties = {
  user_id: number;
  username: string;
};
export type IssueEntity = {
  class: Array<string>;
  rel: Array<string>;
  properties: IssueEntityProperties;
  links: Array<LinksSiren>;
};
export type IssueProperties = {
  user_id: number;
  project_id: number;
  issue_id: number;
  name: string;
  description: string;
  labels: Array<string>;
  opened_on: number;
  closed_on: number;
  state_name: string;
};
export type IssueType = {
  class: Array<string>;
  properties: IssueProperties;
  entities: Array<IssueEntity>;
  actions: Array<ActionsSiren>;
  links: Array<LinksSiren>;
};
